collector/diskstats_common.go

```diff
@@ -72,3 +81,28 @@ var (
 		nil,
 	)
 )
+
+func newDiskstatsDeviceFilter(logger log.Logger) (deviceFilter, error) {
+	if *oldDiskstatsDeviceExclude != "" {
+		if *diskstatsDeviceExclude == "" {
```

Comment:
if you do the opposite here, it will remove an indentation: 

```go
if len(*oldDiskstatsDeviceExclude) > 0 {
        return deviceFilter{}, errors.New("--collector.diskstats.ignored-devices and --collector.diskstats.device-exclude are mutually exclusive")
} 

level.Warn(logger).Log("msg", "--collector.diskstats.ignored-devices is DEPRECATED and will be removed in 2.0.0, use --collector.diskstats.device-exclude")
			*diskstatsDeviceExclude = *oldDiskstatsDeviceExclude
```

Reply:
Again, this is the style / pattern used in the other collectors. I'd like to keep it the way it is.

CC @roidelapluie